public/app/features/alerting/unified/components/rules/RulesTable.tsx

```diff
@@ -1,61 +1,73 @@
-import { GrafanaTheme, rangeUtil } from '@grafana/data';
+import { GrafanaTheme } from '@grafana/data';
 import { ConfirmModal, useStyles } from '@grafana/ui';
-import { CombinedRuleGroup, RulesSource } from 'app/types/unified-alerting';
 import React, { FC, Fragment, useState } from 'react';
 import { getRuleIdentifier, isAlertingRule, stringifyRuleIdentifier } from '../../utils/rules';
 import { CollapseToggle } from '../CollapseToggle';
 import { css, cx } from '@emotion/css';
-import { TimeToNow } from '../TimeToNow';
 import { StateTag } from '../StateTag';
 import { RuleDetails } from './RuleDetails';
 import { getAlertTableStyles } from '../../styles/table';
 import { ActionIcon } from './ActionIcon';
 import { createExploreLink } from '../../utils/misc';
 import { getRulesSourceName, isCloudRulesSource } from '../../utils/datasource';
-import { RulerRuleDTO } from 'app/types/unified-alerting-dto';
 import { useDispatch } from 'react-redux';
 import { deleteRuleAction } from '../../state/actions';
 import { useHasRuler } from '../../hooks/useHasRuler';
+import { CombinedRule } from 'app/types/unified-alerting';
 
 interface Props {
-  namespace: string;
-  group: CombinedRuleGroup;
-  rulesSource: RulesSource;
+  rules: CombinedRule[];
+  showGuidelines?: boolean;
+  showGroupColumn?: boolean;
+  emptyMessage?: string;
 }
 
-export const RulesTable: FC<Props> = ({ group, rulesSource, namespace }) => {
-  const { rules } = group;
+export const RulesTable: FC<Props> = ({
+  rules,
+  showGuidelines = false,
+  emptyMessage = 'No rules found.',
+  showGroupColumn = false,
+}) => {
   const dispatch = useDispatch();
 
-  const hasRuler = useHasRuler(rulesSource);
+  const hasRuler = useHasRuler();
 
   const styles = useStyles(getStyles);
   const tableStyles = useStyles(getAlertTableStyles);
 
   const [expandedKeys, setExpandedKeys] = useState<string[]>([]);
 
-  const [ruleToDelete, setRuleToDelete] = useState<RulerRuleDTO>();
+  const [ruleToDelete, setRuleToDelete] = useState<CombinedRule>();
 
   const toggleExpandedState = (ruleKey: string) =>
     setExpandedKeys(
       expandedKeys.includes(ruleKey) ? expandedKeys.filter((key) => key !== ruleKey) : [...expandedKeys, ruleKey]
     );
 
   const deleteRule = () => {
```

Comment:
Question: why not just dispatch the `deleteRuleAction` on click instead of doing it on next render after we set the state? Probably me that is missing something.

Reply:
setting `ruleToDelete` triggers a confirmation modal to be shown, but does not disaptch the delete action.
delete action is dispatched once user clicks confirmation button on the modal

Reply:
ah, ok ok - got it! Makes sense now.